pkg/proxy/winkernel/proxier.go

```diff
@@ -173,15 +196,97 @@ func conjureMac(macPrefix string, ip net.IP) string {
 	return "02-11-22-33-44-55"
 }
 
-func newEndpointInfo(ip string, port uint16, isLocal bool, hns HostNetworkService) *endpointsInfo {
+func (proxier *Proxier) endpointsMapChange(oldEndpointsMap, newEndpointsMap proxy.EndpointsMap) {
+	for svcPortName := range oldEndpointsMap {
+		proxier.onEndpointsMapChange(&svcPortName)
+	}
+
+	for svcPortName := range oldEndpointsMap {
+		proxier.onEndpointsMapChange(&svcPortName)
+	}
+}
+
+func (proxier *Proxier) onEndpointsMapChange(svcPortName *proxy.ServicePortName) {
+
+	svc, exists := proxier.serviceMap[*svcPortName]
+
+	if exists {
+		svcInfo, ok := svc.(*serviceInfo)
+
+		if !ok {
+			klog.Errorf("Failed to cast serviceInfo %q", svcPortName.String())
+			return
+		}
+
+		klog.V(2).Infof("Endpoints are modified. Service [%v] is stale", *svcPortName)
+		svcInfo.cleanupAllPolicies(proxier.endpointsMap[*svcPortName])
+	} else {
+		// If no service exists, just cleanup the remote endpoints
+		klog.V(2).Infof("Endpoints are orphaned. Cleaning up")
+		// Cleanup Endpoints references
+		epInfos, exists := proxier.endpointsMap[*svcPortName]
+
+		if exists {
+			// Cleanup Endpoints references
+			for _, ep := range epInfos {
+				epInfo, ok := ep.(*endpointsInfo)
+
+				if ok {
+					epInfo.Cleanup()
+				}
+
+			}
+		}
+	}
+}
+
+func (proxier *Proxier) serviceMapChange(previous, current proxy.ServiceMap) {
+	for svcPortName := range current {
+		proxier.onServiceMapChange(&svcPortName)
+	}
+
+	for svcPortName := range previous {
+		if _, ok := current[svcPortName]; ok {
+			continue
+		}
+		proxier.onServiceMapChange(&svcPortName)
+	}
+}
+
+func (proxier *Proxier) onServiceMapChange(svcPortName *proxy.ServicePortName) {
+
+	svc, exists := proxier.serviceMap[*svcPortName]
+
+	if exists {
+		svcInfo, ok := svc.(*serviceInfo)
+
+		if !ok {
+			klog.Errorf("Failed to cast serviceInfo %q", svcPortName.String())
+			return
+		}
+
+		klog.V(1).Infof("Updating existing service port %q at %s:%d/%s", svcPortName, svcInfo.ClusterIP(), svcInfo.Port(), svcInfo.Protocol())
```

Comment:
maybe a slightly lower log level here, either 2 or 3.
```suggestion
		klog.V(2).Infof("Updating existing service port %q at %s:%d/%s", svcPortName, svcInfo.ClusterIP(), svcInfo.Port(), svcInfo.Protocol())
```

Reply:
changed to 3